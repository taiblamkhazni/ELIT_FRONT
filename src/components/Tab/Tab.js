/**
 * @file Tab.js
 * @brief Styled div component for tab interface with conditional styling based on active state
 */
/**
 * @brief styled : import styled-components library
 */
import styled from "styled-components";

/**
 * @brief Styled div component representing a tab.
 * @param align - The alignment of the tab.
 * @param padding - The padding of the tab.
 * @param margin - The margin of the tab.
 * @param cursor - The cursor style of the tab.
 * @param font - The font size of the tab.
 * @param line - The line height of the tab.
 * @param color - The color of the tab.
 * @param border - The border of the tab.
 */
export const Tab = styled.div`
  display: flex;
  align-items: center;
  padding: 1px 3px 0 3px;
  margin-right: 16px;
  cursor: pointer;

  font-size: 18px;
  font-weight: 400;
  line-height: 40px;
  color: ${({ $isactive }) => ($isactive ? "#116E9C" : "#6a6a6a")};
  border-bottom: 2px solid
  ${({ $isactive }) => ($isactive ? "#116E9C" : "transparent")};
`;
