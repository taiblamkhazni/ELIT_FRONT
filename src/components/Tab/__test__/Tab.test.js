/**
 * @file Tab.test.js
 * @brief Ce fichier contient des tests pour les composant Tab.
 */
import renderer from "react-test-renderer";

import { render } from '@testing-library/react'

import { Tab } from "../Tab";

import "jest-styled-components"

test("Tab tests props active", () => {

    const tree = renderer.create(<Tab />).toJSON();

    expect(tree).toMatchSnapshot()
    expect(tree).toHaveStyleRule("border-bottom", "2px solid transparent")
    expect(tree).toHaveStyleRule("color", "#6a6a6a")
})

test('with props', () => {
    const { container } = render(<Tab $isactive={true} />)
    expect(container.firstChild).toMatchSnapshot()
    expect(container.firstChild).toHaveStyleRule("border-bottom", "2px solid #116E9C")
    expect(container.firstChild).toHaveStyleRule("color", "#116E9C")
})