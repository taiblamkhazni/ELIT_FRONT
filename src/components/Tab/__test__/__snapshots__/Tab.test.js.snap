// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tab tests props active 1`] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  padding: 1px 3px 0 3px;
  margin-right: 16px;
  cursor: pointer;
  font-size: 18px;
  font-weight: 400;
  line-height: 40px;
  color: #6a6a6a;
  border-bottom: 2px solid transparent;
}

<div
  className="c0"
/>
`;

exports[`with props 1`] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  padding: 1px 3px 0 3px;
  margin-right: 16px;
  cursor: pointer;
  font-size: 18px;
  font-weight: 400;
  line-height: 40px;
  color: #116E9C;
  border-bottom: 2px solid #116E9C;
}

<div
  class="c0"
/>
`;
